### PR TITLE
Provide v2 role support for `scim2/bulk` API

### PR DESCRIPTION
**Pull request by PasinduYeshan**
## Purpose
$subject

## Related PRs
- https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/pull/491

### CODE REVIEW

File: modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/BulkRequestProcessor.java [hunk 1]
```diff
@@ -254,6 +305,16 @@ private String extractIDFromPath(String path) throws BadRequestException {
         }
     }
 
+    private String extractIDFromV2Path(String path) throws BadRequestException {
+        String [] parts = path.split("[/]");
```
Comment (AnuradhaSK) [inline modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/BulkRequestProcessor.java:309]: missing new line after the method signature

Comment (PasinduYeshan) [reply modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/BulkRequestProcessor.java:310]: Fixed

File: modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/BulkRequestProcessor.java [hunk 2]
```diff
@@ -254,6 +305,17 @@ private String extractIDFromPath(String path) throws BadRequestException {
...
 
+    private String extractIDFromV2Path(String path) throws BadRequestException {
+
+        String [] parts = path.split("[/]");
+        if (parts[3] != null) {
+            return parts[3];
+        } else {
+            throw new BadRequestException
+                    ("No resource Id is provided in path", ResponseCodeConstants.INVALID_VALUE);
+        }
```
Comment (AnuradhaSK) [inline modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/BulkRequestProcessor.java:316]:
```suggestion
        }
        throw new BadRequestException
                    ("No resource Id is provided in path", ResponseCodeConstants.INVALID_VALUE);
```

**Review by AnuradhaSK — DISMISSED** [review]

**Review by AnuradhaSK — APPROVED** [review]

**Comment by PasinduYeshan** [comment]
PR builder failed due to the following intermittent issue.

```
[ERROR] Tests run: 2068, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 6,685.518 s <<< FAILURE! - in TestSuite
2023-10-24T05:44:01.2890376Z [ERROR] testSAMLSSOLogin(org.wso2.identity.integration.test.saml.RegistryMountTestCase)  Time elapsed: 7.416 s  <<< FAILURE!
2023-10-24T05:44:01.2892850Z java.lang.AssertionError: SAML SSO Login failed for travelocity.com-registrymount expected:<true> but was:<false>
2023-10-24T05:44:01.2894632Z 	at org.wso2.identity.integration.test.saml.RegistryMountTestCase.testSAMLSSOLogin(RegistryMountTestCase.java:172)
2023-10-24T05:44:01.2895717Z 
2023-10-24T05:44:03.0360612Z [INFO] 
2023-10-24T05:44:03.0361367Z [INFO] Results:
2023-10-24T05:44:03.0362271Z [INFO] 
2023-10-24T05:44:03.0362637Z [ERROR] Failures: 
2023-10-24T05:44:03.0364268Z [ERROR]   RegistryMountTestCase.testSAMLSSOLogin:172 SAML SSO Login failed for travelocity.com-registrymount expected:<true> but was:<false>
```

File: modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/BulkRequestProcessor.java [hunk 3]
```diff
@@ -161,6 +187,18 @@ public BulkResponseData processBulkRequests(BulkRequestData bulkRequestData) thr
                 }
             }
         }
+        // Handle v2 role operations.
+        for (BulkRequestContent bulkRequestContent : bulkRequestData.getRoleV2OperationRequests()) {
+            if (failOnError == 0) {
+                bulkResponseData.addRoleOperation(getBulkResponseContent(bulkRequestContent, userIdMappings,
+                        roleResourceV2Manager));
+            } else {
```
Comment (sadilchamishka) [inline modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/BulkRequestProcessor.java:195]: We can have else-if

Comment (sadilchamishka) [reply modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/BulkRequestProcessor.java:195]: Update the license header, when you send the commit.

Comment (PasinduYeshan) [reply modules/charon-core/src/main/java/org/wso2/charon3/core/protocol/BulkRequestProcessor.java:195]: Addressed with https://github.com/wso2/charon/pull/407.

**Pull request merged**